js/media.js

```diff
@@ -46,6 +46,10 @@ jQuery( function ( $ ) {
 		},
 
 		remove: function ( models, options ) {
+			// Don't remove models if event came from a Gutenberg component.
+			if( $( event.target ).hasClass( 'components-button' ) || $( event.target ).parents().hasClass( 'components-button' ) ) {
+				return;
+			}
```

Comment:
I'm a little bit confused here since the `event` variable is undefined.

Reply:
I was a bit confused by the same thing since `media.js` is using jQuery strict mode. After a bit of research I found that `event` is essentially a global variable that resolves to `window.event`.

**Here are some links with more info:**
1. https://api.jquery.com/category/events/event-object/
2. https://developer.mozilla.org/en-US/docs/Web/API/Window/event
3. https://stackoverflow.com/questions/6426497/is-event-a-global-variable-that-is-accessible-everywhere-inside-the-callback-cha

I'm sure this isn't the best overall way to handle this, but it's a quick option with minimal overall code edits. There may be inconsistencies across browsers, but I haven't noticed any yet.

**Here is another reference that touches on a similar situation:**
https://stackoverflow.com/questions/25380310/how-to-determine-why-the-remove-event-of-backbone-collection-is-triggered

Reply:
Interesting read. I didn't know `event` is global.